File: src/test/java/com/mikea/bayes/belief/ClusterGraphsTest.java
package com.mikea.bayes.belief;

import com.mikea.bayes.BayesianNetwork;
import com.mikea.bayes.StudentsNetwork;
import org.junit.Test;

/**
 * @author mike.aizatsky@gmail.com
 */
public class ClusterGraphsTest {
    @Test
    public void testValidate() throws Exception {
        BayesianNetwork network = StudentsNetwork.buildStudentsNetwork();
        ClusterGraph graph = CliqueTree.buildCliqueTree(network);

        ClusterGraphs.validate(graph, network.getFactorList());
    }
}

File: src/test/java/com/mikea/bayes/belief/ClusterGraphsTest.scala
package com.mikea.bayes.belief

import com.mikea.bayes.BayesianNetwork
import com.mikea.bayes.data.StudentsNetwork
import org.junit.Test

/**
 * @author mike.aizatsky@gmail.com
 */
class ClusterGraphsTest {
  @Test def testValidate {
    val network: BayesianNetwork = StudentsNetwork.buildStudentsNetwork
    val graph: ClusterGraph = CliqueTree.buildCliqueTree(network)
    ClusterGraphs.validate(graph, network.getFactorList)
  }
}